The first two hunks make sure only qmake is build, as the rest of Qt is built
in other ports.

The last hunk is backported from upstream:

commit a9474d1260a8c8cc9eae14f2984098919d9684e5
Author: Raphael Kubo da Costa <rakuco@FreeBSD.org>
Date:   Tue Aug 23 20:25:14 2016 +0200

    configure: Correctly detect clang's version on FreeBSD

    "clang -v" on FreeBSD reports something like "FreeBSD clang version
    x.y.z [...]" instead of just "clang version x.y.z [...]", which fails to
    match the sed pattern in the configure script, resulting in qconfig.pri
    having no clang version defined.

    Augment the pattern so that both version strings match.

    Change-Id: I5f38f8480f4b1156ca7147e32c1157a009557035
    Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>

--- configure.orig	2016-08-24 08:18:37 UTC
+++ configure
@@ -4115,11 +4115,11 @@ if [ '!' -e "$outpath/bin/qmake" ]; then
         echo "QT_MAJOR_VERSION = $QT_MAJOR_VERSION" >> "$mkfile"
         echo "QT_MINOR_VERSION = $QT_MINOR_VERSION" >> "$mkfile"
         echo "QT_PATCH_VERSION = $QT_PATCH_VERSION" >> "$mkfile"
-        echo "EXTRA_CFLAGS = $EXTRA_CFLAGS" >> "$mkfile"
-        echo "EXTRA_CXXFLAGS = $EXTRA_CXXFLAGS" >> "$mkfile"
+        echo "EXTRA_CFLAGS = $EXTRA_CFLAGS $CFLAGS" >> "$mkfile"
+        echo "EXTRA_CXXFLAGS = $EXTRA_CXXFLAGS $CXXFLAGS" >> "$mkfile"
         echo "QTOBJS =" $EXTRA_OBJS >> "$mkfile"
         echo "QTSRCS =" $EXTRA_SRCS >> "$mkfile"
-        echo "LFLAGS = $EXTRA_LFLAGS" >> "$mkfile"
+        echo "LFLAGS = $EXTRA_LFLAGS $LDFLAGS" >> "$mkfile"
         echo "EXEEXT = $EXEEXT" >> "$mkfile"
         echo "RM_F = rm -f" >> "$mkfile"
         echo "RM_RF = rm -rf" >> "$mkfile"
@@ -4138,6 +4138,7 @@ if [ '!' -e "$outpath/bin/qmake" ]; then
             fi
         fi
     done
+    exit 0
 
     if [ "$OPT_VERBOSE" = yes ]; then
         # Show the output of make
@@ -6691,7 +6689,7 @@ case "$QMAKE_CONF_COMPILER" in
     # Clang
     COMPILER_VERSION=`${QMAKE_CONF_COMPILER} -v 2>&1 | sed -n -E '
 /^Apple (clang|LLVM) version /{s///; s/^([0-9]*)\.([0-9]*).*$/QT_APPLE_CLANG_MAJOR_VERSION=\1; QT_APPLE_CLANG_MINOR_VERSION=\2/;p;q;}
-/^clang version /{s///; s/^([0-9]*)\.([0-9]*).*$/QT_CLANG_MAJOR_VERSION=\1; QT_CLANG_MINOR_VERSION=\2/;p;q;}'`
+/^(FreeBSD )?clang version /{s///; s/^([0-9]*)\.([0-9]*).*$/QT_CLANG_MAJOR_VERSION=\1; QT_CLANG_MINOR_VERSION=\2/;p;q;}'`
     eval "$COMPILER_VERSION"
     ;;
 *icpc)
